pkg/storage/storage.go

```diff
@@ -1,68 +1,28 @@
 package storage
 
 import (
-	"fmt"
-	"io"
 	"time"
 
+	"github.com/tilezen/tapalcatl/pkg/state"
 	"github.com/tilezen/tapalcatl/pkg/tile"
 )
 
 type Storage interface {
-	Fetch(t tile.TileCoord, c Condition, prefixOverride string) (*StorageResponse, error)
-	TileJson(f TileJsonFormat, c Condition, prefixOverride string) (*StorageResponse, error)
+	Fetch(t tile.TileCoord, c state.Condition, prefixOverride string) (*StorageResponse, error)
+	TileJson(f state.TileJsonFormat, c state.Condition, prefixOverride string) (*StorageResponse, error)
 	HealthCheck() error
 }
 
-type Condition struct {
-	IfModifiedSince *time.Time
-	IfNoneMatch     *string
-}
-
 type SuccessfulResponse struct {
-	Body         io.ReadCloser
+	Body         []byte
```

Comment:
The data read from S3 and from the Metatile used to be in buffers, but I switched them to `[]byte` so they can be more easily cached.

Reply:
nice